### DISCUSSION TITLE
Environment variables as placeholders

### DISCUSSION THREAD

**Discussion by cascandaliato** [flo-at/minmon#193] [Ideas]
I'm using `minmon` through Docker and some of the actions are [Pushover](https://pushover.net/) webhooks that require API keys. I'd like to keep my Pushover keys in a `.env` file and inject them into the container environment.

**Comment by flo-at** [comment]
That would be a good option to manage secrets. The syntax could be something like `{{env:env_var_name}}`.
The env. var. name should have a (configurable?) common prefix so we don't have to store all env. vars. (because that can be a lot).

**Reply by cascandaliato** [reply]
> The env. var. name should have a (configurable?) common prefix

Good point. Even a non-configurable prefix like `MINMON_` sounds reasonable enough to me.

Related to the use case of secrets, I would also consider allowing placeholders in a Webhook `headers` (possibly a different discussion/issue).

**Comment by flo-at** [comment]
Check out [v0.9.0](https://github.com/flo-at/minmon/releases/tag/v0.9.0). Feel free to reopen if something is missing or not working as it should.

**Reply by cascandaliato** [reply]
Thank you for implementing my request so quickly.

I tested few cases and they all worked perfectly:
- env vars in webhook header
- env vars in webhook body
- default, custom non-empty and custom empty (i.e. import all vars) env var prefix

**Reply by flo-at** [reply]
Thanks for testing
